File: projects/ios/Sample/Shared/Models/SimpleOption.swift
import Foundation
import UIKit

struct SimpleOption: Hashable {
    let type: OptionTypeEnum
    let hasSeparator: Bool

    func getDescription() -> String {
        switch type {
        case .secretKey:
            return "OptionSecretKey".localized
        case .sharedData:
            let demoFlag = EZRHelperSharedDataHelper.getDemoFlag()
            return String(format: "%@ %@", "OptionSharedData".localized, demoFlag ? "ON" : "OFF")
        case .httpRequest:
            return "OptionHttpRequest".localized
        case .httpsRequest:
            return "OptionHttpsRequest".localized
        case .fileHelper:
            return "OptionFileHelper".localized
        case .appVersion:
            return "OptionAppVersion".localized
        case .todo:
            return "OptionTodo".localized
        case .webServer:
            let demoFlag = EZRHttpServer.shared()?.isRunning() ?? false
            return String(format: "%@ %@", "OptionWebServer".localized, demoFlag ? "ON" : "OFF")
        case .webView:
            return "OptionWebView".localized
        }
    }

    func getImage() -> UIImage {
        return UIImage(named: "IcoSimpleOption")!.imageWithColor(color: UIColor(hexString: "#D21601")!)!
    }
}

File: projects/android/Sample/app/src/main/java/com/ezored/sample/model/SimpleOption.kt
package com.ezored.sample.model

import android.content.Context
import com.ezored.sample.R
import com.ezored.sample.enumerator.SimpleOptionTypeEnum

class SimpleOption(val type: SimpleOptionTypeEnum) {

    fun getDescription(context: Context): String {
        return when {
            type === SimpleOptionTypeEnum.SHARED_DATA -> context.getString(R.string.option_shared_data)
            type === SimpleOptionTypeEnum.HTTPS_REQUEST -> context.getString(R.string.option_https_request)
            type === SimpleOptionTypeEnum.SECRET_KEY -> context.getString(R.string.option_secret_key)
            type === SimpleOptionTypeEnum.APP_VERSION -> context.getString(R.string.option_app_version)
            type === SimpleOptionTypeEnum.TODO -> context.getString(R.string.option_todo)
            type === SimpleOptionTypeEnum.FILE_HELPER -> context.getString(R.string.option_file_helper)
            type === SimpleOptionTypeEnum.WEB_SERVER -> context.getString(R.string.option_web_server)
            type === SimpleOptionTypeEnum.WEB_VIEW -> context.getString(R.string.option_web_view)
            else -> ""
        }
    }

    fun getImage(): Int {
        return R.drawable.ic_simple_option
    }
}
